classes/note/NoteDAO.inc.php

```diff
@@ -122,6 +127,34 @@ public function getByAssoc($assocType, $assocId, $userId = null, $orderBy = self
         return new DAOResultFactory($result, $this, '_fromRow', [], $sql, $params); // Counted in QueriesGridCellProvider
     }
 
+    /**
+     * Retrieve all Notes by assoc id/type
+     *
+     * @param int $assocId ASSOC_TYPE_...
+     * @param int $assocType Assoc ID (per $assocType)
+     * @param int $userId Optional user ID
+     *
+     * @return LazyCollection of Notes
+     */
+    public function getAllByAssoc($assocType, $assocId, $userId = null)
```

Comment:
At a glance, this new function isn't necessary -- getByAssoc should already be enough.

Reply:
Here I created the new method for 2 main reasons:
1. This code deals with Notes that could have both their `title` and `contents` columns have `null` values. Those cases will not be retrieved by the `getByAssoc` method because of the `AND (title IS NOT NULL OR contents IS NOT NULL)` clause.
2. I didn't want, for this case, to order the data. That's less important of course.

I could perhaps change the `getByAssoc` method to include as a parameter the use of `AND (title IS NOT NULL OR contents IS NOT NULL)` with a flag or something, but I thought it would be safer to not mess with that.

Reply:
I think I'd prefer extending the old function to add the new option. Better still -- though there's some scope creep here -- would be to rewrite the old function to use the new query construction tools. I'd like for us to do this gradually when there's opportunity.

Reply:
@asmecher I refactored the `getByAssoc` to the query construction tools and added a new switch for sorting data (or not). It seems that `isAdmin` switch was enough for the use of `AND (title IS NOT NULL OR contents IS NOT NULL)` or not. 

I have pushed the changes and the tests on OJS, OMP, OPS have started.